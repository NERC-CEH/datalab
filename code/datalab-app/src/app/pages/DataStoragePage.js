import React from 'react';
import Typography from 'material-ui/Typography';
import Segment from '../components/app/Segment';
import DataStorageContainer from '../containers/dataStorage/DataStorageContainer';

const DataStoragePage = () => (
  <Segment>
    <Typography gutterBottom type="display1">Current Storage Volumes</Typography>
    <DataStorageContainer />
  </Segment>
);

export default DataStoragePage;
