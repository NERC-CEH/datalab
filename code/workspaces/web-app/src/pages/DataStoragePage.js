import React from 'react';
import PropTypes from 'prop-types';
import DataStorageContainer from '../containers/dataStorage/DataStorageContainer';
import Page from './Page';

const DataStoragePage = ({ userPermissions, match }) => (
  <Page title="Storage">
    <DataStorageContainer userPermissions={userPermissions} projectKey={match.params.projectKey} />
  </Page>
);

DataStorageContainer.propTypes = {
  userPermissions: PropTypes.arrayOf(PropTypes.string).isRequired,
  projectKey: PropTypes.string.isRequired,
};

export default DataStoragePage;
