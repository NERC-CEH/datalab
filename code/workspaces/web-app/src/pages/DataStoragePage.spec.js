import React from 'react';
import { render } from '@testing-library/react';
import DataStoragePage from './DataStoragePage';

const userPermissions = ['expectedPermission'];

jest.mock('../containers/dataStorage/DataStorageContainer', () => props => (<div>DataStorageContainer mock {JSON.stringify(props)}</div>));
jest.mock('../components/app/Footer', () => () => (<div>Footer mock</div>));

it('DataStoragePage renders correct snapshot', () => {
  expect(render(<DataStoragePage userPermissions={userPermissions} />).container).toMatchSnapshot();
});
