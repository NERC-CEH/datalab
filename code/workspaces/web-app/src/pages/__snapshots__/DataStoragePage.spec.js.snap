// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataStoragePage renders correct snapshot 1`] = `
<WithStyles(Page)
  title="Storage"
>
  <Connect(DataStorageContainer)
    projectKey="project99"
    userPermissions={
      Array [
        "expectedPermission",
      ]
    }
  />
</WithStyles(Page)>
`;
